uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeaveRequestTest.php

```diff
@@ -4071,6 +4082,177 @@ public function testLeaveRequestBeCreatedWhenLeaveRequestHasNoWorkingDayAndValid
+      'start_date' => CRM_Utils_Date::processDate('2018-01-01'),
+      'end_date'   => CRM_Utils_Date::processDate('2018-12-31'),
+    ]);
+
+    $periodEntitlement = LeavePeriodEntitlementFabricator::fabricate([
+      'type_id' => $absenceType->id,
+      'contact_id' => $contactId,
+      'period_id' => $absencePeriod->id
+    ]);
+
+    $this->createLeaveBalanceChange($periodEntitlement->id, 3);
+
+    $params = [
+      'type_id' => $absenceType->id,
+      'contact_id' => $contactId,
+      'from_date' => CRM_Utils_Date::processDate($date . ' 01:00:00'),
+      'to_date' => CRM_Utils_Date::processDate($date . ' 03:00:00')
+    ];
+
+    $leaveRequest1 = LeaveRequestFabricator::fabricate($params, true);
+
+    $params['from_date'] = CRM_Utils_Date::processDate($date . ' 03:00:00');
+    $params['to_date'] = CRM_Utils_Date::processDate($date . ' 05:00:00');
+
+    $leaveRequest2 = LeaveRequestFabricator::fabricate($params, true);
+
+    $this->assertNotNull($leaveRequest1->id);
+    $this->assertNotNull($leaveRequest2->id);
+  }
+
+  public function testTOILAccrualAndLeaveRequestCanBeCreatedForTheSameDay() {
+    $date = '2018-04-13';
+    $contactId = 1;
+    $absenceTypeLeave = AbsenceTypeFabricator::fabricate();
+    $absenceTypeTOIL = AbsenceTypeFabricator::fabricate([
+      'allow_accruals_request' => true,
+      'allow_accrue_in_the_past' => true
+    ]);
+
+    $absencePeriod = AbsencePeriodFabricator::fabricate([
+      'start_date' => CRM_Utils_Date::processDate('2018-01-01'),
+      'end_date'   => CRM_Utils_Date::processDate('2018-12-31'),
+    ]);
+
+    $periodEntitlementLeave = LeavePeriodEntitlementFabricator::fabricate([
+      'type_id' => $absenceTypeLeave->id,
+      'contact_id' => $contactId,
+      'period_id' => $absencePeriod->id
+    ]);
+
+    $periodEntitlementTOIL = LeavePeriodEntitlementFabricator::fabricate([
+      'type_id' => $absenceTypeTOIL->id,
+      'contact_id' => $contactId,
+      'period_id' => $absencePeriod->id
+    ]);
+
+    HRJobContractFabricator::fabricate(
+      ['contact_id' => $contactId],
+      ['period_start_date' => CRM_Utils_Date::processDate('2018-01-01')]
+    );
+
+    $this->createLeaveBalanceChange($periodEntitlementLeave->id, 3);
+    $this->createLeaveBalanceChange($periodEntitlementTOIL->id, 3);
+
+    $params = [
+      'type_id' => $absenceTypeLeave->id,
+      'contact_id' => $contactId,
+      'from_date' => CRM_Utils_Date::processDate($date),
+      'to_date' => CRM_Utils_Date::processDate($date)
+    ];
+
+    $leaveRequest1 = LeaveRequestFabricator::fabricate($params, true);
+
+    $params['request_type'] = LeaveRequest::REQUEST_TYPE_TOIL;
+    $params['type_id'] = $absenceTypeTOIL->id;
+
+    $leaveRequest2 = LeaveRequestFabricator::fabricate($params, true);
+
+    $this->assertNotNull($leaveRequest1->id);
+    $this->assertNotNull($leaveRequest2->id);
+  }
+
+  public function testTwoLeaveRequestsOneAMAndAnotherPMCanBeCreatedForTheSameDay() {
```

Comment:
What about another test to show the inverse of this test, i.e that two leave requests with same from_date_type and to_date type cannot be created on the same day? i.e a test for criteria 4 `It should not be possible to create more than one AM leave request within the same day. Same for PM.`

Reply:
Also I've not seen any test for criteria 3 on the PR description i.e `It should not be possible to create one leave request in days and one leave request in hours on the same day, regardless of the request in hours start and end time.`

Reply:
Also the test for inverse of criteria 6: `It should be possible for multiple TOIL requests to be created for one day as long as their start-end time does not overlap.` i.e a test to show that an exception will be thrown when creating a TOIL request with overlapping datetime.

Reply:
Done.